feat: nav optimization

berry-13 opened the pull request:
## Overview

This PR is a major overhaul of the conversations history, search, and related navigation systems. It introduces modern infinite/cursor-based pagination for all conversations and search endpoints, leverages virtualization for large lists, and resolves longstanding data consistency and UX issues. The scope is both backend and frontend

---

### Key Features and Changes

#### :rocket: Infinite Scroll & Cursor-Based Pagination

- **Backend**
  - `/api/convos` and `/api/search` endpoints now implement cursor-based pagination (using `nextCursor`/`updatedAt`), allowing for fast and reliable infinite scrolling
  - All relevant API endpoints and model methods refactored for cursor support, improved filtering, error handling, and consistency
  - Search results are returned in the same paginated format, compatible with infinite scroll loading

- **Frontend**
  - Conversations list in the sidebar is now powered by [react-virtualized List](https://github.com/bvaughn/react-virtualized), enabling smooth rendering of large numbers of conversations
  - Fetching logic now uses React Query's infiniteQuery, cursor keys, and updated utilities for cache consistency
  - All add, edit (rename, archive), and delete operations properly update/invalidate infinite paginated caches
  - UX supports loading spinners, robust error handling, and instant feedback

#### :mag: Search Overhaul

- Unified the search UX for both navigation and dedicated search page
- Typing in the search bar is debounced, showing precise loading indicators and never blocks UI
- Results are fetched and paginated infinitely, consistent with main conversation listing
- Enhanced search message display with additional metadata and consistent formatting

#### :bookmark: Archive & Restore Improvements

- Archived conversations are now managed and displayed via the same infinite/cursor-based system as main history
- Unarchiving and deleting now provide immediate feedback and do not require full page reloads
- Improved dialogs and error notifications for destructive/archive actions

#### :wrench: Code Quality, Maintainability, and UI Details

- Major cleanup and refactor of utilities for managing infinite/cursor-based conversation sets
- Extracted and modernized components such as RenameForm, ConvoLink
- Unified button style and accessibility across navigation UI
- Removed deprecated/unused code and test files
- Updated translation keys for better user messages (e.g. "Untitled", error handling, etc.)

---

### Breaking Changes

- Pagination APIs for `/api/convos` and `/api/search` no longer use page numbers, but instead use a cursor (`nextCursor`) mechanism
- Any integrations/scripts relying on old page-based navigation must be updated

---

**This is a major step forward in performance, scalability, and UX for conversation management and search in LibreChat!**

(description made with AI)

## Change Type

- [x] a11y
- [x] Performance

## Testing

Tested on Chrome, Firefox and Brave for both Desktop and Mobile

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.

berry-13 committed:
✨ feat: improve Nav/Conversations/Convo/NewChat component performance

berry-13 committed:
✨ feat: improve Nav/Conversations/Convo/NewChat component performance

berry-13 committed:
✨ feat: implement cursor-based pagination for conversations API

berry-13 committed:
🔧 refactor: remove createdAt from conversation selection in API and type definitions

berry-13 committed:
🔧 refactor: include createdAt in conversation selection and update related types

berry-13 committed:
✨ fix: search functionality and bugs with loadMoreConversations

berry-13 committed:
feat: move ArchivedChats to cursor and DataTable standard

berry-13 committed:
🔧 refactor: add InfiniteQueryObserverResult type import in Nav component

berry-13 committed:
feat: enhance conversation listing with pagination, sorting, and search capabilities

berry-13 committed:
🔧 refactor: remove unnecessary comment regarding lodash/debounce in ArchivedChatsTable

berry-13 committed:
🔧 refactor: remove unnecessary comment regarding lodash/debounce in ArchivedChatsTable

berry-13 committed:
🔧 refactor: remove unused translation keys for archived chats and search results

berry-13 committed:
🔧 fix: Archived Chats, Delete Convo, Duplicate Convo

berry-13 committed:
🔧 refactor: improve conversation components with layout adjustments and new translations

berry-13 committed:
🔧 refactor: simplify archive conversation mutation and improve unarchive handling; fix: update fork mutation

berry-13 committed:
🔧 refactor: decode search query parameter in conversation route; improve error handling in unarchive mutation; clean up DataTable component styles

berry-13 committed:
🔧 refactor: remove unused translation key for empty archived chats

danny-avila committed:
🧠 feat: Bedrock Anthropic Reasoning & Update Endpoint Handling (#6163)

* feat: Add thinking and thinkingBudget parameters for Bedrock Anthropic models

* chore: Update @librechat/agents to version 2.1.8

* refactor: change region order in params

* refactor: Add maxTokens parameter to conversation preset schema

* refactor: Update agent client to use bedrockInputSchema and improve error handling for model parameters

* refactor: streamline/optimize llmConfig initialization and saving for bedrock

* fix: ensure config titleModel is used for all endpoints

* refactor: enhance OpenAIClient and agent initialization to support endpoint checks for OpenRouter

* chore: bump @google/generative-ai

berry-13 committed:
🚀 fix: `archivedConversation` query key not updated correctly while archiving

berry-13 committed:
🔧 refactor: update translation keys for clarity; simplify conversation query parameters and improve sorting functionality in SharedLinks component

berry-13 committed:
🔧 refactor: optimize conversation loading logic and improve search handling in Nav component

berry-13 committed:
fix: package-lock

berry-13 committed:
fix: package-lock 2

berry-13 committed:
fix: package lock 3

berry-13 committed:
refactor: remove unused utility files and exports to clean up the codebase

berry-13 committed:
refactor: remove i18n and useAuthRedirect modules to streamline codebase

berry-13 committed:
refactor: optimize Conversations component and remove unused ToggleContext

berry-13 committed:
refactor(Convo): add RenameForm and ConvoLink components; enhance Conversations component with responsive design

danny-avila commented on api/package.json:
```diff
@@ -39,7 +39,6 @@
     "@aws-sdk/s3-request-presigner": "^3.758.0",
     "@azure/identity": "^4.7.0",
     "@azure/search-documents": "^12.0.0",
-    "@azure/storage-blob": "^12.26.0",
```
this dependency was removed, intentional or mistake when updating the branch?

berry-13 committed:
fix: add missing @azure/storage-blob dependency in package.json

berry-13 committed:
refactor(Search): add error handling with toast notification for search errors

danny-avila committed:
refactor: make createdAt and updatedAt fields of tConvoUpdateSchema less restrictive if timestamps are missing

danny-avila committed:
chore: update @azure/storage-blob dependency to version 12.27.0, ensure package-lock is correct

berry-13 committed:
refactor(Search): improve conversation handling server side

berry-13 committed:
fix: eslint warning and errors

berry-13 committed:
refactor(Search): improved search loading state and overall UX

berry-13 committed:
Refactors conversation cache management

Centralizes conversation mutation logic into dedicated utility functions for adding, updating, and removing conversations from query caches.

Improves reliability and maintainability by:
- Consolidating duplicate cache manipulation code
- Adding type safety for infinite query data structures
- Implementing consistent cache update patterns across all conversation operations
- Removing obsolete conversation helper functions in favor of standardized utilities

berry-13 committed:
fix: conversation handling and SSE event processing

- Optimizes conversation state management with useMemo and proper hook ordering
- Improves SSE event handler documentation and error handling
- Adds reset guard flag for conversation changes
- Removes redundant navigation call
- Cleans up cursor handling logic and document structure

Improves code maintainability and prevents potential race conditions in conversation state updates

berry-13 committed:
refactor: add type for SearchBar `onChange`

berry-13 committed:
fix: type tags

berry-13 committed:
style: rounded to xl all Header buttons

berry-13 committed:
fix: activeConvo in Convo not working

berry-13 committed:
style(Bookmarks): improved UI

berry-13 committed:
a11y(AccountSettings): fixed hover style not visible when using light theme

berry-13 committed:
style(SettingsTabs): improved tab switchers and dropdowns

berry-13 committed:
feat: add translations keys for Speech

berry-13 committed:
chore: fix package-lock

berry-13 committed:
fix(mutations): legacy import after rebase

berry-13 committed:
feat: refactor conversation navigation for accessibility

berry-13 committed:
fix(search): convo and message create/update date not returned

berry-13 committed:
fix(search): show correct iconURL and endpoint for searched messages

berry-13 committed:
fix: small UI improvements

berry-13 committed:
chore: console.log cleanup

berry-13 committed:
chore: fix tests

danny-avila committed:
fix(ChatForm): improve conversation ID handling and clean up useMemo dependencies

danny-avila committed:
chore: improve typing

danny-avila committed:
chore: improve typing

danny-avila committed:
fix(useSSE): clear conversation ID on submission to prevent draft restoration

danny-avila committed:
refactor(OpenAIClient): clean up abort handler

danny-avila committed:
refactor(abortMiddleware): change handleAbort to use function expression

danny-avila committed:
feat: add PENDING_CONVO constant and update conversation ID checks

danny-avila committed:
fix: final event handling on abort

danny-avila committed:
fix: improve title sync and query cache sync on final event

danny-avila committed:
fix: prevent overwriting cached conversation data if it already exists

danny-avila commented:
### Other Changes added:
- Fix an issue with conversation "folding" introduced by recent optimizations
- Properly cleanup abort signal listener and allow aborting requests (was broken by recent optimizations)
- Added a "PENDING" `conversationId` to improve UI/UX with auto-saved drafts persisting immediately after submission
